src/transformers/modeling_tapas.py

```diff
@@ -0,0 +1,2265 @@
+
+
+@add_start_docstrings(
+    """
+    Tapas Model with a cell selection head and optionally aggregation head on top for question-answering tasks on
+    tables (linear layers on top of the hidden-states output to compute `logits` and optionally `logits_aggregation`),
+    e.g. for SQA, WTQ or WikiSQL tasks.
+    """,
+    TAPAS_START_DOCSTRING,
+)
+class TapasForQuestionAnswering(TapasPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        # base model
+        self.tapas = TapasModel(config)
+
+        # dropout (only used when training)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+        # cell selection heads
+        if config.init_cell_selection_weights_to_zero:
+            # init_cell_selection_weights_to_zero: Whether the initial weights should be
+            # set to 0. This ensures that all tokens have the same prior probability.
+            self.output_weights = nn.Parameter(torch.zeros(config.hidden_size))
+            self.column_output_weights = nn.Parameter(torch.zeros(config.hidden_size))
+        else:
+            self.output_weights = nn.Parameter(torch.empty(config.hidden_size))
+            nn.init.normal_(
+                self.output_weights, std=0.02
+            )  # here, a truncated normal is used in the original implementation
+            self.column_output_weights = nn.Parameter(torch.empty(config.hidden_size))
+            nn.init.normal_(
+                self.column_output_weights, std=0.02
+            )  # here, a truncated normal is used in the original implementation
+        self.output_bias = nn.Parameter(torch.zeros([]))
+        self.column_output_bias = nn.Parameter(torch.zeros([]))
+
+        # aggregation head
+        if config.num_aggregation_labels > 0:
+            self.aggregation_classifier = nn.Linear(config.hidden_size, config.num_aggregation_labels)
+
+        self.init_weights()
+
+    @add_start_docstrings_to_model_forward(TAPAS_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
+    @replace_return_docstrings(output_type=TableQuestionAnsweringOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids=None,
+        attention_mask=None,
+        token_type_ids=None,
+        position_ids=None,
+        head_mask=None,
+        inputs_embeds=None,
+        table_mask=None,
+        label_ids=None,
+        aggregation_labels=None,
+        float_answer=None,
+        numeric_values=None,
+        numeric_values_scale=None,
+        output_attentions=None,
+        output_hidden_states=None,
+        return_dict=None,
+    ):
+        r"""
+        table_mask (:obj:`torch.LongTensor` of shape :obj:`(batch_size, seq_length)`, `optional`):
+            Mask for the table. Indicates which tokens belong to the table (1). Question tokens, table headers and
+            padding are 0.
+        label_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, seq_length)`, `optional`):
```

Comment:
we usually have the input `labels` -> could we maybe use this one here as well instead of `label_ids`?

Reply:
The reason these are called `label_ids` is that one of the 7 token type ids which the authors of TAPAS define is called `prev_label_ids`. These indicate which tokens were part of an answer to the previous table-question pair, which is useful in a conversational set-up. Using `_ids` at the end reflects that these are similar to `input_ids` (also having shape `(batch_size, sequence_length)`, but are zero or one depending on whether a token is (part of) an answer. So IMO it's more clear to use `label_ids` rather than `labels`, otherwise people might be confused when they see `prev_label_ids` in the docs. 

Labels which are really similar to other `labels` in the Transformers library are the `aggregation_labels` to train `TapasForQuestionAnswering` and `labels` to train `TapasForSequenceClassification`.

Reply:
From reading the description, I still think that `label_ids` is used to compute a loss. Most other `labels` in the library also have the shape `(batch_size, seq_length)` and there is no `_ids` in the end. But I'm not enough into the code to fully understand what's going on...maybe @LysandreJik can decide here :-)

Reply:
We renamed this to labels ;) cc @patrickvonplaten